src/DynamoCoreWpf/Views/FileTrust/FileTrustWarning.xaml.cs

```diff
@@ -47,6 +47,15 @@ public FileTrustWarning(DynamoView dynamoViewWindow)
             SetUpPopup();
 
             HomeWorkspaceModel.WorkspaceClosed += CloseWarningPopup;
+            dynViewModel.PropertyChanged += DynViewModel_PropertyChanged;
+        }
+
+        private void DynViewModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "CurrentSpace" && !((sender as DynamoViewModel).HomeSpaceViewModel as HomeWorkspaceViewModel).IsCurrentSpace)
```

Comment:
hmm why not just watch ViewingHomespace? This seems to be the same check

Reply:
Done